JsonSchema.Generation/SchemaGeneratorContext.cs

```diff
@@ -24,6 +25,91 @@ public ContextCount(SchemaGeneratorContext context)
 			}
 		}
 
+		private class MemberInfoMetadataTokenComparer : Comparer<MemberInfo>
+		{
+
+			private readonly int[] _typeOrder;
+
+			private MemberInfoMetadataTokenComparer(Type type)
+			{
+				var typeStack = new Stack<Type>();
+
+				do
+				{
+					typeStack.Push(type);
+					type = type.BaseType!;
+				} while (type != null);
+
+				_typeOrder = typeStack.Select(GetMetadataToken).ToArray();
+			}
+
+			public static MemberInfoMetadataTokenComparer ForType(Type type)
+			{
+				return new MemberInfoMetadataTokenComparer(type ?? throw new ArgumentNullException(nameof(type)));
+			}
```

Comment:
For internal things, I think it's okay to just expose the constructor.  I don't see much benefit in this pattern when it just creates a new one anyway.